### ISSUE TITLE
About SDAR Training Setup

### ISSUE THREAD

**Issue by clf28** [JetAstra/SDAR#12]
Thank you for your brilliant work！
I noticed you mentioned SDAR has been trained in a 2-stage way on pretrained Qwen3 series in Sec5.1.1. I have two questions for that:
1. Is the training rule following the description in Sec3.1: "The entire sequence is perturbed during the pretraining stage, whereas during SFT, the prompt remains clean and the response is perturbed." ? What will happen if skipping the stage1, because what we need is only to adapt the attention direction and loss convergence way.
2. Is all models with different size all trained with 50B data in total, even for 1.7B model?

I would be very greatful if getting the reply

**Comment by unsosp1ro** [comment]
1. It's possible to skip continued pretraining(CPT) and SFT directly, but we found that the performance scores are generally lower. We guess that, with higher-quality and larger-scale training data, along with careful hyperparameter tuning, direct SFT perhaps achieves better results. 
2. Yes, all models except for `SDAR-30B-A3B-Sci` are continued pretrained on the same 50B-token data.

**Comment by clf28** [comment]
Great! May I ask when will the training code release? Is the training framework based on DiffuLLaMA, or other codebase?

**Comment by clf28** [comment]
Great! Thank you for your reply.
